lib/cluster/s3_cold_archive_bucket.rb

```diff
@@ -0,0 +1,90 @@
+module Cluster
+  class S3ColdArchiveBucket < S3DistributionBucket
+    def self.find_or_create(bucket_name)
+      bucket = S3Bucket.find_existing(name: bucket_name)
+      return bucket if bucket
+
+      S3Bucket.create(name: bucket_name)
+
+      # Currently, there's no way to specify object minimum size in the filter
```

Comment:
Reference: https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Types/LifecycleRuleFilter.html